src/libsync/utility_win.cpp

```diff
@@ -15,21 +15,34 @@
 #include <shlobj.h>
 #include <winbase.h>
 #include <windows.h>
+#include <shlguid.h>
+#include <string>
+#include <QLibrary>
 
 static const char runPathC[] = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Run";
 
+
+     
 static void setupFavLink_private(const QString &folder)
 {
     // Windows Explorer: Place under "Favorites" (Links)
-    wchar_t path[MAX_PATH];
-    SHGetSpecialFolderPath(0, path, CSIDL_PROFILE, FALSE);
-    QString profile =  QDir::fromNativeSeparators(QString::fromWCharArray(path));
+    
+    QString linkName; 
     QDir folderDir(QDir::fromNativeSeparators(folder));
-    QString linkName = profile+QLatin1String("/Links/") + folderDir.dirName() + QLatin1String(".lnk");
+     
+    /* Use new WINAPI functions */
+    wchar_t *path = NULL;
+    if(SHGetKnownFolderPath(FOLDERID_Links, 0, NULL, &path) == S_OK) {
```

Comment:
From the documentation about the `path` argument:

> The calling process is responsible for freeing this resource once it is no longer needed by calling CoTaskMemFree.

Could you verify or fix it?